armi/reactor/tests/test_blocks.py

```diff
@@ -395,6 +395,21 @@ def test_getSmearDensityMultipleLiner(self):
         ) / self.block.getDim(Flags.INNER | Flags.LINER, "id") ** 2
         self.assertAlmostEqual(cur, ref, places=10)
 
+    def test_getSmearDensityEdgeCases(self):
+        # show smear density is not computed for non-fuel blocks
+        b0 = blocks.Block("DummyReflectorBlock")
+        self.assertEqual(b0.getSmearDensity(), 0.0)
+
+        # show smear density is only defined for pinned fuel blocks
+        b1 = blocks.HexBlock("TestFuelHexBlock")
+        b1.setType("fuel")
+        b1.p.nPins = 0
+        fuel = components.Circle(
+            "fuel", "UZr", Tinput=25.0, Thot=25.0, od=0.84, id=0.6, mult=0
+        )
+        b1.add(fuel)
+        self.assertEqual(b1.getSmearDensity(), 0.0)
```

Comment:
This test is confusing until you catch the `nPins = 0` line. I see a circle of UZr, that looks like a pin and we should have a non-zero smear density.

What about making it a hex of sodium? Just to make it even more clear there should be no fuel nor fuel pins in the block?

Reply:
Oh, wait.

Actually, the second "if" statement I am testing is that the it IS a fuel block, but it ISN'T pinned. So, the block needs to be defined as "fuel", but not "pinned", that's the edge case this PR is about.

Reply:
Oooooo gotcha gotcha. Yep, that makes sense.